### DISCUSSION TITLE
Support for variables

### DISCUSSION THREAD

**Discussion by mike-solomon** [GitbookIO/community#374] [Feature Requests]
## Context

One feature I really wish GitBook would support is variables. We have a variety of guides that include code snippets that specify versions of various libraries. It is a pretty huge pain trying to keep those versions up-to-date so that people coming to the docs don't have to figure out what the current versions are. It's possible people may end up in a state where their code is broken because we were late in updating some of these versions. Trying to do a search and replace is often impossible as many libraries have similar version numbers.

## Desired outcome

Ideally, I would be able to define some variable such as `$VERSION_REWRITE_SPRING` and a value for it such as `5.1.11`. Then, I could use this variable throughout many docs. When the version changes, I would only need to update the variable file rather than every doc. I could, theoretically, even make some automation to connect to GitHub to detect the latest release and automatically update this file (which would then update all of my docs).

It would be even more lovely if GitBook offered this functionality where I could point to a repository and it could keep the versions up-to-date by looking at the releases.

## What's the impact of this for you?

We currently have thousands of docs in our GitBook site. This is ramping up quickly over time and the site is becoming harder to maintain because we have no way of automating updates throughout. Versions and code blocks are core to our docs.

- [ ] Critical
- [x] High
- [ ] Medium
- [ ] Low

## Related issue

I found an [open issue about snippets](https://github.com/orgs/GitbookIO/discussions/121) that is somewhat similar to this. It's not quite the same, but I felt like it was worth linking to.

**Comment by biccio** [comment]
This would be incredibly helpful 🙏🏻

**Reply by tomorrowx-git** [reply]
💯 agree @biccio last reply I had... 10 Jan '24 doesn't look like it's got enough support...yet

_Hey Stuart,
Thank you for your message.
Sadly, we don't have that feature built to GitBook.
This is something we'd like to explore in the future, but we don't have any timeframe for that to happen at this time.
Hopefully that helps clarify your question.
Have a wonderful day,
Rodrigo_

> I see I'm not the only one asking for this... https://github.com/orgs/GitbookIO/discussions/374


_Hey Stuart,
Yes, this is something we've heard from other customers as well, which is why we'd like to explore that route in the future.
But unfortunately, as I said, we don't have a timeframe to share when this will happen.
So, for now, there's not much we can provide here.
Thank you,
Rodrigo_

**Comment by MadeByPP** [comment]
I would very much like to use unix time stamp. In the case of Discord, it can be used through [time stamp style](https://discord.com/developers/docs/reference#message-formatting-timestamp-styles). Of course, some documents may be sufficient for the last edit time to appear at the end of the document, but others may not.

**Comment by drewsth** [comment]
Is there any update and/or plan for this feature, to make use of variables?
I'm currently thinking about getting additional documents converted/included within gitbook and this feature would addresses a key need for these documents and would therefore make a very positive impact on the workflow.

**Comment by philipp-schmidt** [comment]
+1 for this feature
